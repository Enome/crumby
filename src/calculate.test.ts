import { describe, it, expect } from "vitest";
import subject from "./calculate";

describe("calculate", () => {
  it("calculates the output", () => {
    const state = {
      name: "Recipe",
      doughballCount: 2,
      doughballWeight: 850,
      percentages: [
        { id: 1, name: "Flour", percentage: 100.0, base: true },
        { id: 2, name: "Water", percentage: 67.0, base: true },
        { id: 3, name: "Salt", percentage: 2.0, base: true },
        { id: 4, name: "Yeast", percentage: 0.05, base: true },
      ],
      flours: [
        { id: 1, name: "White", percentage: 90.0 },
        { id: 2, name: "Wholemeal", percentage: 10.0 },
      ],
      sourdough: true,
      preferment: true,

      sourdoughTarget: 12.0,
      sourdoughRatio: "1:2:2",

      prefermentType: "Poolish",
      prefermentTarget: 12.0,
      prefermentRatio: "1:2",
    };

    expect(subject(state)).toMatchSnapshot();
  });
});
