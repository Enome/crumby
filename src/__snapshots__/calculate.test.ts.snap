// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calculate > calculates the output 1`] = `
{
  "dough": [
    {
      "base": true,
      "id": 1,
      "name": "Flour",
      "percentage": 100,
      "weight": 835.6196391600117,
    },
    {
      "base": true,
      "id": 2,
      "name": "Water",
      "percentage": 67,
      "weight": 435.76515823720786,
    },
    {
      "base": true,
      "id": 3,
      "name": "Salt",
      "percentage": 2,
      "weight": 20.112392783200235,
    },
  ],
  "flourWeight": 1005.6196391600117,
  "flours": [
    {
      "id": 1,
      "name": "White",
      "percentage": 90,
      "weight": 905.0576752440106,
    },
    {
      "id": 2,
      "name": "Wholemeal",
      "percentage": 10,
      "weight": 100.56196391600118,
    },
  ],
  "floursWeight": 1005.6196391600118,
  "percentagesAndWeights": [
    {
      "base": true,
      "id": 1,
      "name": "Flour",
      "percentage": 100,
      "weight": 1005.6196391600117,
    },
    {
      "base": true,
      "id": 2,
      "name": "Water",
      "percentage": 67,
      "weight": 673.7651582372079,
    },
    {
      "base": true,
      "id": 3,
      "name": "Salt",
      "percentage": 2,
      "weight": 20.112392783200235,
    },
    {
      "base": true,
      "id": 4,
      "name": "Yeast",
      "percentage": 0.05,
      "weight": 0.5028098195800059,
    },
  ],
  "preferment": {
    "flour": 68,
    "total": 204.50280981958,
    "water": 136,
    "yeast": 0.5028098195800059,
  },
  "sourdough": {
    "flour": 81.6,
    "starter": 40.8,
    "total": 204,
    "water": 81.6,
  },
  "targetWeight": 1700,
  "totalPercentagesWeight": 1700,
}
`;
